optimade/server/main.py

```diff
@@ -91,30 +106,153 @@ def update_schema(app):
         json.dump(app.openapi(), f, indent=2)
 
 
+def general_exception(
+    request: Request, exc: Exception, **kwargs: Dict[str, Any]
+) -> JSONResponse:
+    tb = "".join(
+        traceback.format_exception(etype=type(exc), value=exc, tb=exc.__traceback__)
+    )
+    print(tb)
+
+    try:
+        status_code = exc.status_code
+    except AttributeError:
+        status_code = kwargs.get("status_code", 500)
+
+    try:
+        detail = exc.detail
+    except AttributeError:
+        detail = str(exc)
+
+    errors = kwargs.get("errors", None)
+    if not errors:
+        errors = [
+            Error(detail=detail, status=status_code, title=str(exc.__class__.__name__))
+        ]
+
+    return JSONResponse(
+        status_code=status_code,
+        content=jsonable_encoder(
+            ErrorResponse(
+                meta=meta_values(
+                    str(request.url), 0, 0, False, **{CONFIG.provider + "traceback": tb}
+                ),
+                errors=errors,
+            ),
+            skip_defaults=True,
+        ),
+    )
+
+
+@app.exception_handler(StarletteHTTPException)
+def http_exception_handler(request: Request, exc: Exception):
```

Comment:
Type of `exc` can be more accurate. Both for this and the other exception handlers.